src/robot/running/testlibraries.py

```diff
@@ -213,7 +214,12 @@ def _try_to_create_handler(self, name, method):
             self._report_adding_keyword_failed(name)
 
     def _create_handler(self, handler_name, handler_method):
-        return Handler(self, handler_name, handler_method)
+        handler = Handler(self, handler_name, handler_method)
+        if '$' in handler.name:
+            embedded = EmbeddedArguments(handler.name)
+            if embedded:
+                handler = EmbeddedArgsTemplate(embedded, handler)
+        return handler
```

Comment:
This logic should be moved to `_create_handlers` (or a helper method used by it) in order to avoid duplicating the exact same logic in `_DynamicLibrary._create_handler`.